enterprise_access/apps/bffs/api.py

```diff
@@ -70,10 +84,111 @@ def get_and_cache_enterprise_customer(
         enterprise_customer_uuid=enterprise_customer_uuid,
         enterprise_customer_slug=enterprise_customer_slug,
     )
-    TieredCache.set_all_tiers(cache_key, response_payload, settings.LMS_CLIENT_TIMEOUT)
+    TieredCache.set_all_tiers(cache_key, response_payload, timeout)
+    return response_payload
+
+
+def get_and_cache_subscription_licenses_for_learner(
+    request,
+    enterprise_customer_uuid,
+    timeout=settings.SUBSCRIPTION_LICENSES_LEARNER_CACHE_TIMEOUT,
+    **kwargs
+):
+    """
+    Retrieves and caches subscription licenses for a learner.
+    """
+    cache_key = subscription_licenses_cache_key(enterprise_customer_uuid)
+    cached_response = TieredCache.get_cached_response(cache_key)
+    if cached_response.is_found:
+        logger.info(
+            f'subscription_licenses cache hit for enterprise_customer_uuid {enterprise_customer_uuid}'
+        )
+        return cached_response.value
+
+    client = LicenseManagerUserApiClient(request)
+    response_payload = client.get_subscription_licenses_for_learner(
+        enterprise_customer_uuid=enterprise_customer_uuid,
+        **kwargs,
+    )
+    TieredCache.set_all_tiers(cache_key, response_payload, timeout)
     return response_payload
 
 
+def get_and_cache_default_enterprise_enrollment_intentions(
+    request,
+    enterprise_customer_uuid,
+    timeout=settings.DEFAULT_ENTERPRISE_ENROLLMENT_INTENTIONS_CACHE_TIMEOUT,
+):
+    """
+    Retrieves and caches default enterprise enrollment intentions for a learner.
+    """
+    cache_key = default_enterprise_enrollment_intentions_cache_key(enterprise_customer_uuid)
+    cached_response = TieredCache.get_cached_response(cache_key)
+    if cached_response.is_found:
+        logger.info(
+            f'default_enterprise_enrollment_intentions cache hit '
+            f'for enterprise_customer_uuid {enterprise_customer_uuid}'
+        )
+        return cached_response.value
+
+    client = LmsUserApiClient(request)
+    response_payload = client.get_default_enterprise_enrollment_intentions_learner_status(
+        enterprise_customer_uuid=enterprise_customer_uuid,
+    )
+    TieredCache.set_all_tiers(cache_key, response_payload, timeout)
+    return response_payload
+
+
+def get_and_cache_enterprise_course_enrollments(
+    request,
+    enterprise_customer_uuid,
+    timeout=settings.ENTERPRISE_COURSE_ENROLLMENTS_CACHE_TIMEOUT,
+    **kwargs
+):
+    """
+    Retrieves and caches enterprise course enrollments for a learner.
+    """
+    cache_key = enterprise_course_enrollments_cache_key(enterprise_customer_uuid)
```

Comment:
This cache key should take the requesting user's id into account, too.

Reply:
Ah, yes. Even though the underlying request doesn't pass a user identified (i.e., relying on the request user), the cache key definitely needs to take it into account! Great catch.